client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx

```diff
@@ -127,14 +172,19 @@ const PersonalInfoTab: React.FC<Props> = ({ id, onSubmit }: Props): JSX.Element
         setAddress(address);
     }, [data.city, data.street, data.floor, data.houseNum]);
 
-    React.useEffect(() => {
-        if(investigationId !== -1) {
+    useEffect(() => {
+        roleObj = investigatedPatientRoles.find((investigatedPatientRole: investigatedPatientRole) => investigatedPatientRole.id === data.role);
```

Comment:
Cant you use useMemo to update roleObj and than listen to it on useEffect to change the input?